File: Frontend/src/models/product/product-create-dto.ts
import { ProductImageCreateDto } from "./product-image-create-dto";
import { ProductVariantCreateDto } from "./product-variant-create-dto";

export interface ProductCreateDto {
  name: string;
  description: string;
  price: number;
  brand: string;
  categoryId: number;
  variants: ProductVariantCreateDto[];
  images: ProductImageCreateDto[];
}

File: Backend/ClothingStore.Core/Models/Product/ProductCreateDTO.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClothingStore.Core.Models.Product
{
    public class ProductCreateDTO
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        [Precision(18, 2)]
        public decimal Price { get; set; }

        public string Brand { get; set; } = string.Empty;

        public int CategoryId { get; set; }

        public List<ProductVariantCreateDTO> Variants { get; set; } = new();

        public List<ProductImageCreateDto> Images { get; set; } = new();
    }
}
